samples/net/bm_serial_test/prj.conf

```diff
@@ -26,4 +26,7 @@ CONFIG_UART_PIPE=y
 CONFIG_NET_CONFIG_IEEE802154_DEV_NAME="IEEE802154_UPIPE"
 CONFIG_BM_SERIAL_DEV_NAME_0="UART_2"
 CONFIG_BM_SERIAL_DEV_NAME_1=""
-CONFIG_BM_SERIAL_DEV_NAME_2=""
\ No newline at end of file
+CONFIG_BM_SERIAL_DEV_NAME_2=""
+
+CONFIG_UART_BM=y
+CONFIG_UART_ASYNC_API=y
```

Comment:
How about we update the Kconfig files and/or driver to make it so that UART_ASYNC_API is either no longer required (ideal) or such that selecting UART_BM automatically selects UART_ASYNC_API. Since uart_bm depends on DMA and the async events, its use should imply the use of any required dependencies.

Reply:
This would likely be done in drivers/serial/Kconfig.bm

Reply:
It looks like certain structs defined outside the scope of this driver depend on the UART_ASYNC_API being set. I can go with the latter approach of automatically selecting this configuraiton